src/BuildScriptGeneratorCli/Options/PhpScriptGeneratorOptionsSetup.cs

```diff
@@ -0,0 +1,30 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.BuildScriptGenerator;
+using Microsoft.Oryx.BuildScriptGenerator.Php;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli.Options
+{
+    /// <summary>
+    /// Gets hierarchical configuration from IConfiguration api and
+    /// binds the properties on <see cref="PhpScriptGeneratorOptions"/>.
+    /// </summary>
+    public class PhpScriptGeneratorOptionsSetup : IConfigureOptions<PhpScriptGeneratorOptions>
+    {
+        private readonly IEnvironment _environment;
+
+        public PhpScriptGeneratorOptionsSetup(IEnvironment environment)
```

Comment:
I am sending this PR now because I will later be sending a PR where we would be following `Microsoft.Extensions.Configuration` api to get hierarchical configuration from different 'sources' (command line -> environment variables -> files like build.env and in future maybe app.yml). 
So instead of reading from `IEnvironment` directly here, we would be injecting `IConfiguration` and do something like `configuation["PHP_VERSION"]` where this `configuration` will take care of 'resolving' the final value based on the hierarchically set values.

Reply:
is that other PR needs to go first before this one? or it  does not matter?

Reply:
I am not too concerned about the order at this point since `build.env` has not been advertised yet and also my PR would be coming out soon.